back/app/controllers/application_controller.rb

```diff
@@ -103,4 +103,11 @@ def paginate(collection)
     collection.page(params.dig(:page, :number))
       .per(params.dig(:page, :size))
   end
+
+  def remove_image_if_requested!(resource, resource_params, image_field_name)
+    return unless resource_params.key?(image_field_name) && resource_params[image_field_name].nil?
```

Comment:
It varies from form to form. Sometimes the param has nil, and sometimes there's no key in params at all.

Anyway, I tried not to change the behavior in all the cases, so it should work as before.

> If it's the first one, we would always attempt to delete that file, even if it has never been set I think.

I believe it's still fine.

Reply:
* in all the cases apart from folders and phases.

Reply:
Okay then.

Do you think it makes sense to move the logic to the model instead? I think it fits a bit better than having the ApplicationController contain the logic of image attachment deletion. We could use `ActiveModel::Dirty` for this:

```
ruby
class User
  def remove_image_file_when_nil
    avatar.remove_image if avatar.changed? && avatar.nil?
  end
end
```

or generalize it a bit more and add it to the `ApplicationRecord` so it works with any kind of image attachment on any model.

Reply:
I thought about it, but I am not sure it doesn't break anything. Because there can be so many variations of changes. I assumed the risk isn't worth it.